src/IISIntegration/src/Microsoft.AspNetCore.Server.IIS/Core/IISHttpContext.cs

```diff
@@ -258,7 +258,21 @@ private async Task ProduceStart(bool flushHeaders)
                 }
             }
 
-            _writeBodyTask = WriteBody(!flushHeaders);
+            if (!canHaveNonEmptyBody)
+            {
+                _bodyOutput.Dispose();
+            }
+            else
+            {
+                _writeBodyTask = WriteBody(!flushHeaders);
+            }
+        }
+
+        private bool StatusCodeCanHaveBody()
+        {
+            return StatusCode != 204
+                && StatusCode != 205
```

Comment:
The rules for 205 are quite different:
```
   Since the 205 status code implies that no additional content will be
   provided, a server MUST NOT generate a payload in a 205 response.  In
   other words, a server MUST do one of the following for a 205
   response: a) indicate a zero-length body for the response by
   including a Content-Length header field with a value of 0; b)
   indicate a zero-length payload for the response by including a
   Transfer-Encoding header field with a value of chunked and a message
   body consisting of a single chunk of zero-length; or, c) close the
   connection immediately after sending the blank line terminating the
   header section.
```
As opposed to 204:
```
 A 204 response is terminated by the first empty line after the header
   fields because it cannot contain a message body.
```

Reply:
In this context you should remove 205 from the list.

Reply:
Kestrel also checks for 205. Should it remove the check too?

Reply:
https://github.com/aspnet/AspNetCore/blob/master/src/Servers/Kestrel/Core/src/Internal/Http/HttpProtocol.cs#L1202

Reply:
Likely, but you can file a separate issue to address that.